lib/ruby_terraform/command_line/option.rb

```diff
@@ -0,0 +1,60 @@
+require 'json'
+
+module RubyTerraform
+  module CommandLine
+    class Option
+      def initialize(opt_key, value, switch_override: nil)
+        assign_option_value(value)
+        @switch = switch_override || switch_from_key(opt_key)
+      end
+
+      def add_to_subcommand(sub)
+        if value.respond_to?(:keys)
+          add_hash_to_subcommand(sub, value)
+        elsif value.respond_to?(:each)
+          add_array_to_subcommand(sub, value)
+        else
+          add_option_to_subcommand(sub)
+        end || sub
```

Comment:
What's the `|| sub` for here? Since the `if` has an `else` I can't see that it will ever be reached.

Reply:
hang over from some old code.. really truly was an rfc for the overall concept of using declarative commands rather than seeking a full code review..  I'll make sure whatever I put up for a real PR is a lot cleaner :)